plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -631,10 +631,15 @@ static obs_properties_t *vaapi_properties(void *unused)
 	if (obs_property_list_item_count(list) == 0) {
 		char path[32] = "/dev/dri/renderD1";
 		for (int i = 28;; i++) {
-			sprintf(path, "/dev/dri/renderD1%d", i);
+			snprintf(path, sizeof(path), "/dev/dri/renderD1%d", i);
 			if (access(path, F_OK) == 0) {
 				char card[128] = "Card: ";
```

Comment:
Initializing this seems weird when we immediately overwrite it.

Reply:
Changed both `snprintf` invocations to use a `const char` template string and also removed the initial string assignment (as it's immediately overwritten anyway as you mentioned).

Reply:
I'm personally not a fan of moving the format strings away from the `snprintf` invocation as it adds additional programmer overhead to parse what it's doing, but this solves my main concern.

Reply:
I wouldn't be opposed to it, I tried to surmise why it was written that way to begin with and thought the idea was to have a single "template" string that needs to be adjusted, but that train of thought also doesn't make sense.